### DISCUSSION TITLE
How to use fetch during startup in nuxt.config.ts?

### DISCUSSION THREAD

**Discussion by niklasfjeldberg** [nuxt/nuxt#23519] [Questions]
I would like to fetch data from an external API during start up in the `nuxt.config.ts` file. Is this possible? The [Meilisearch module](https://nuxt.com/modules/meilisearch) needs two keys, which can be gotten by `GET` at `[Meilisearch domain].com/keys.` I would like to use fetch to get the keys and use these them in the configuration instead of environmental variables:

```
meilisearch: {
    hostUrl: process.env.MEILI_HOST,
    searchApiKey: process.env.MEILI_SEARCH_API_KEY,
    adminApiKey: process.env.MEILI_ADMIN_API_KEY,
  },
```

I am able to get the keys with node fetch, but not to use them. Any ideas?

```
const key = async () => {
  const response = await fetch('https://domain.com/keys', {
    method: 'GET',
    headers: { 'Authorization': 'Bearer [KEY]' }
  });
  const jsonResponse = await response.json();

  let searchApiKey: string;
  let adminApiKey: string;

  if (jsonResponse && jsonResponse.results) {
    for (const key of jsonResponse.results) {
      console.log(key.name);
      console.log(key.key);
      if (key.name === 'Default Search API Key') {
        searchApiKey = key.key;
      } else if (key.name === 'Default Admin API Key') {
        adminApiKey = key.key;
      }
    }
  }
  return {
    searchApiKey,
    adminApiKey
  };
};
```

This gives me undefined variables when trying to use them. How do I make the code wait until the variables are populated?

**Comment by xlanex6** [comment]
Hi 

If you `needs` to fetch key you should explore runtime hooks to inject key for later use. 

more info https://nuxt.com/docs/guide/going-further/runtime-config
but 
If you can avoid this over complex process, grab your key manuel, but maybe you *have to *....

**Comment by or2e** [comment]
@niklasfjeldberg 
Consider this approach: https://nuxt.com/modules/prepare
